Frameworks/Foundation/NSURLResponse.mm

```diff
@@ -89,8 +94,22 @@ - (id)_initWithURL:(NSURL*)url
         _url.attach([url copy]);
         _mimeType.attach([mimeType copy]);
         _textEncodingName.attach([textEncodingName copy]);
-        if (nil != fileName) {
-            _suggestedFilename.attach([fileName copy]);
+
+        if (fileName && ![fileName isEqualToString:s_unknownFileName]) {
+            woc::unique_cf<CFStringRef> identifier {
+                _UTTypeCreatePreferredIdentifierForTag(static_cast<CFStringRef>(@"public.mime-type"),
+                                                       static_cast<CFStringRef>(mimeType),
+                                                       nullptr)
+            };
+            StrongId<NSString> extension {
+                static_cast<NSString*>(
+                    _UTTypeCopyPreferredTagWithClass(identifier.get(), static_cast<CFStringRef>(@"public.filename-extension")))
+            };
```

Comment:
well that way to initialize variables looks absurdly confusing to me. Why did you choose multiline brace init that makes it look like a function definition?